src/core/reactor.cc

```diff
@@ -1172,7 +1172,10 @@ cpu_stall_detector_posix_timer::cpu_stall_detector_posix_timer(cpu_stall_detecto
     struct sigevent sev = {};
     sev.sigev_notify = SIGEV_THREAD_ID;
     sev.sigev_signo = signal_number();
-    sev._sigev_un._tid = syscall(SYS_gettid);
+#ifndef sigev_notify_thread_id
```

Comment:
You forgot to mention in the commit message or a comment that what Musl (and FreeBSD, and the Linux kernel) have a sigev_notify_thread_id macro. Only glibc is missing it (https://sourceware.org/bugzilla/show_bug.cgi?id=27417) so we used that internal thing.

Reply:
okay, will add that